add solid to pages framework guides (#7442)
cloudflare/cloudflare-docs: content/pages/framework-guides/deploy-a-solid.md

---
pcx_content_type: how-to
title: Deploy a Solid site
---

# Deploy a Solid Site

[Solid](https://www.solidjs.com/) is an open-source web application framework focused on generating performant applications with a modern developer experience based on JSX.

In this guide, you will create a new Solid application implemented via [SolidStart](https://start.solidjs.com/getting-started/what-is-solidstart) (Solid's meta-framework) and deploy it using Cloudflare Pages.

{{<Aside type="warning" header="Important">}}

> At the time of writing SolidStart is in beta, this may cause things to change and make the following guide inaccurate.

Reviewer: ```suggestion
At the time of writing SolidStart is in beta, this may make the following guide inaccurate in the event of a breaking change.
```

{{</Aside>}}

## Create a new project

> Create a new project by running the following command in your terminal:

Reviewer: ```suggestion
Create a new project (`my-solid-app`) by running the following command in your terminal:
```

In order to use SolidStart with Cloudflare Pages you'll add the [SolidStart Cloudflare Pages adapter](https://github.com/solidjs/solid-start/tree/main/packages/start-cloudflare-pages) to your application, to do so, install the adaptor with:

```sh
$ npm install --save-dev solid-start-cloudflare-pages
```

Next you need to use the adapter in your the vite.config.(ts/js) file:

```diff
---
filename: vite.config.ts
---
import solid from "solid-start/vite";
import { defineConfig } from "vite";
+ import cloudflare from "solid-start-cloudflare-pages";

export default defineConfig({
-  plugins: [solid()],
+  plugins: [solid({ adapter: cloudflare({}) })],
});
```

{{<render file="_tutorials-before-you-start.md">}}

## Creating a GitHub repository

Create a new GitHub repository by visiting [repo.new](https://repo.new). After creating a new repository, prepare and push your local application to GitHub by running the following commands in your terminal:

```sh
$ git init
$ git remote add origin https://github.com/<username>/<repo>
$ git add .
$ git commit -m "Initial commit"
$ git branch -M main
$ git push -u origin main
```

## Deploying with Cloudflare Pages

> Deploy your site to Pages by logging in to the [Cloudflare dashboard](https://dash.cloudflare.com/) > **Account Home** > **Pages** and selecting **Create a project**. Select the new GitHub repository that you created and, in the **Set up builds and deployments** section, provide the following information:

Reviewer: ```suggestion
Deploy your site to Pages by logging in to the [Cloudflare dashboard](https://dash.cloudflare.com/) > **Account Home** > **Pages** and selecting **Create a project**. Select the new GitHub repository that you created and, in **Set up builds and deployments**, provide the following information:
```

<div>

| Configuration option  | Value              |
| --------------------- | ------------------ |
| Production branch     | `main`             |
| Build command         | `npm run build`    |
| Build directory       | `dist/public`      |
| Environment Variables | `NODE_VERSION: 17` |

</div>

After configuring your site, you can begin your first deploy. You should see Cloudflare Pages installing `npm`, your project dependencies, and building your site before deploying it.

{{<Aside type="note">}}

For the complete guide to deploying your first site to Cloudflare Pages, refer to the [Get started guide](/pages/get-started/).

{{</Aside>}}

After deploying your site, you will receive a unique subdomain for your project on `*.pages.dev`.
Every time you commit new code to your Solid repository, Cloudflare Pages will automatically rebuild your project and deploy it. You will also get access to [preview deployments](/pages/platform/preview-deployments/) on new pull requests, to preview how changes look to your site before deploying them to production.

> ## Using bindings in your Solid application

Reviewer: ```suggestion
## Use bindings in your Solid application
```

A [binding](/pages/platform/functions/bindings/) allows your application to interact with Cloudflare developer products, such as [KV](https://developers.cloudflare.com/workers/learning/how-kv-works/), [Durable Object](/workers/learning/using-durable-objects/), [R2](/r2/), and [D1](https://blog.cloudflare.com/introducing-d1/).

> To add a binding in your SolidStart, let's start by adding them to the Cloudflare adapter, this allows you to conveniently access those during development, allowing you to test your application before deploying it.

Reviewer: ```suggestion
To add a binding in your SolidStart, add the binding to the Cloudflare adapter. This allows you to access your bindings during development, and test your application before deploying it.
```

Reviewer: Are they added to the CF adapter by updating the `vite.config` file?

Author: yes, as you can see below with the MY_KV one 🙂

> Let's supposed that you want to create a new binding for a KV named "MY_KV", update the adapter's use in your the vite.config.(ts/js) file, for example in the following way:

Reviewer: ```suggestion
The following example configuration supposes you are using a KV namespace binding named `"MY_KV"`. You will need to update the adapter's use in your `vite.config.(ts/js)` file in the following way:
```

```diff
---
filename: vite.config.ts
---
import solid from "solid-start/vite";
import { defineConfig } from "vite";
import cloudflare from "solid-start-cloudflare-pages";

export default defineConfig({
-  plugins: [solid({ adapter: cloudflare({}) })],
+  plugins: [
+   solid({
+     adapter: cloudflare({
+       kvNamespaces: ["MY_KV"],
+     }),
+   }),
+  ],
});
```

> In your code you can then access the binding in the environment variable provided to you by SolidStart, for example in a [createServerData$](https://start.solidjs.com/api/createServerData) loader:

Reviewer: ```suggestion
You can then access the binding in the environment variable provided to you by SolidStart. For example, in a [`createServerData$`](https://start.solidjs.com/api/createServerData) loader:
```

Reviewer: Is the $ intentional?

Author: yes, it's intentional, as you can see if you follow the link 🙂 

[
![Screenshot 2023-02-08 at 08 57 26](https://user-images.githubusercontent.com/61631103/217482443-556adf75-1901-4197-a9e3-4e8befed358b.png)
](https://start.solidjs.com/api/createServerData)

```ts
export function routeData() {
  return createServerData$(async (_, { env }) => {
    const { MY_KV } = (env as { MY_KV:  KVNamespace }));
    // ...
  });
}
```

{{<Aside type="note">}}

> The Cloudflare adapter accepts the same options [Miniflare](https://miniflare.dev/) does, you can see a reference of the various available options in this [Miniflare API Reference](https://miniflare.dev/get-started/api#reference).

Reviewer: ```suggestion
The Cloudflare adapter accepts the same options as [Miniflare](https://miniflare.dev/). Review the available options in this [Miniflare API Reference](https://miniflare.dev/get-started/api#reference).
```
